### PR TITLE
added new_nonuniform_scaling_wrt_point to Matrix3 & Matrix4

### PR DESCRIPTION
**Pull request by adamnemecek**

### CODE REVIEW

File: src/base/cg.rs [hunk 1]
```diff
@@ -70,6 +71,17 @@ impl<N: RealField> Matrix3<N> {
     pub fn new_rotation(angle: N) -> Self {
         Rotation2::new(angle).to_homogeneous()
     }
+
+    /// Creates a new homogeneous matrix that applies a scaling factor for each dimension with respect to point.
+    ///
+    /// Can be used to implement "zoom_to" functionality.
+    #[inline]
+    pub fn new_nonuniform_scaling_wrt_point(scaling: Vector2<N>, pt: Point2<N>) -> Self {
```
Comment (sebcrozet) [inline src/base/cg.rs:79]: Arguments should be provided by-ref to be coherent with the remaining of the cg api.

```suggestion
    pub fn new_nonuniform_scaling_wrt_point(scaling: &Vector2<N>, pt: &Point2<N>) -> Self {
```

File: src/base/cg.rs [hunk 2]
```diff
@@ -90,6 +102,17 @@ impl<N: RealField> Matrix4<N> {
         Isometry::rotation_wrt_point(rot, pt).to_homogeneous()
     }
 
+    /// Creates a new homogeneous matrix that applies a scaling factor for each dimension with respect to point.
+    ///
+    /// Can be used to implement "zoom_to" functionality.
+    #[inline]
+    pub fn new_nonuniform_scaling_wrt_point(scaling: Vector3<N>, pt: Point3<N>) -> Self {
```
Comment (sebcrozet) [inline src/base/cg.rs:109]:
```suggestion
    pub fn new_nonuniform_scaling_wrt_point(scaling: &Vector3<N>, pt: &Point3<N>) -> Self {
```

**Comment by adamnemecek** [comment]
Note that the formatting of the arguments passed to matrix::new is from cargo fmt.

File: src/base/cg.rs (hunk 1 above)
Comment (adamnemecek) [reply src/base/cg.rs]: Why does matrix3::new_rotation (and others) take the argument by value?

Comment (sebcrozet) [reply src/base/cg.rs]: You are right. This looks inconsistent. We might need to revisit this at some point in the future, and settle in one conversion.

**Comment by sebcrozet** [comment]
Thanks!

**Pull request merged**